### ISSUE TITLE
how do ignore changes for attributes in count resource

### ISSUE THREAD

**Issue by wobeng** [hashicorp/terraform#21941]
```
resource "azurerm_function_app" "functionapp" {
  count = 2
  app_settings ={
  "foo" : "bar",
  "foo2" : "bar2",
  }

  lifecycle {
  ignore_changes = [app_settings.*["foo"]]
  }

}
```

current ignore changes don't work

**Comment by phinze** [comment]
Hi there - It looks like you're trying to ignore changes on the `app_settings["foo"]` sub-key for both resources generated by the `count = 2` - is that correct?

If that's the case, according to the [docs](https://www.terraform.io/docs/configuration/resources.html#ignore_changes) you can omit the `*` and it should work like you want it to.

I'm going to close this tentatively but if I'm misunderstanding feel free to follow up and we can reopen.

**Comment by wobeng** [comment]
It's not working :(. Question...how will you ignore changes on max_vcpus?

```
resource "aws_batch_compute_environment" "encode" {
  compute_environment_name = "encode"

  compute_resources {
    instance_role = aws_iam_instance_profile.aws_batch_ecs_instance_role.arn
    instance_type = [
      "optimal"]
    max_vcpus = 256
    min_vcpus = 0

    security_group_ids = [
      data.aws_security_group.selected.vpc_id]
    subnets = data.aws_subnet_ids.selected.ids
    bid_percentage = 100
    type = "SPOT"
    spot_iam_fleet_role = aws_iam_role.aws_batch_spotfleet_role.arn
  }
  type = "MANAGED"
  service_role = aws_iam_role.aws_batch_service_role.arn
  depends_on = [
    "aws_iam_role_policy_attachment.aws_batch_service_role"]
}

```

**Comment by phinze** [comment]
Hi again - sorry to hear it's not working for you. I'll reopen so we can confirm the behavior.

`ignore_changes = compute_resources[max_vcpus]` should work to ignore changes in `max_vcpus`.

If this doesn't work for you, can you share a complete example so we can reproduce the issue?

**Comment by wobeng** [comment]
@phinze I will try it out..what if there was a count in the resource? will it be the same?

**Comment by phinze** [comment]
Yep that syntax should work to ignore `max_vcpus` changes regardless of what count is set to. Let us know if you can get it working!

**Comment by wobeng** [comment]
I got

```

Error: Invalid expression

  on batch/main.tf line 85, in resource "aws_batch_compute_environment" "encode":
  85:   lifecycle {ignore_changes = compute_resources[max_vcpus]}

A static list expression is required.

```

**Comment by teamterraform** [comment]
Hi @wobeng,

The `ignore_changes` argument expects a list of attributes to ignore, like this:

```hcl
lifecycle {
  ignore_changes = [
    compute_resources[max_vcpus]
  ]
}
```

You used a single attribute directly, rather than a list, which is why Terraform reported that a static list expression is required.

The original error here was fixed in 0.12.3, so we're going to close this now. If you have any follow-up questions about the `ignore_changes` feature, feel free to start a new topic in [the community forum](https://discuss.hashicorp.com/c/terraform-core). Thanks!
